spec/rspec/rails/matchers/active_job_spec.rb

```diff
@@ -457,6 +472,14 @@ def perform; raise StandardError; end
       }.to raise_error(/expected to enqueue exactly 1 jobs, but enqueued 0/)
     end
 
+    it "fails if the arguments do not match the job's signature" do
+      two_args_job.perform_later(1)
+
+      expect {
+        expect(two_args_job).to have_been_enqueued.with(1)
+      }.to raise_error(ArgumentError, /Wrong number of arguments/)
+    end
```

Comment:
It would be great to have a keyword example vs positional arguments as well given the original context.

Reply:
Good point, added additional examples for this